item/views.py

```diff
@@ -0,0 +1,113 @@
+from rest_framework import mixins, viewsets, status
+from django.db.models import Q
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from item.models import Item, Currency, Price
+from item.serializers import (
+    ItemRetrieveSerializer,
+    ItemListSerializer,
+    ItemDeleteSerializer,
+    ItemCreateUpdateSerializer,
+    CurrencyCreateUpdateSerializer,
+    CurrencyDeleteSerializer,
+    CurrencyListSerializer,
+    CurrencyRetrieveSerializer,
+    PriceUpdateSerializer,
+    PriceListSerializer,
+    PriceRetrieveSerializer
+)
+from item.filters import ItemNameFilter, PriceItemFilter
+from user.permissions import BlackListPermission
+
+
+class ItemViewSet(mixins.ListModelMixin,
+                  mixins.RetrieveModelMixin,
+                  mixins.UpdateModelMixin,
+                  mixins.CreateModelMixin,
+                  mixins.DestroyModelMixin,
+                  viewsets.GenericViewSet):
+    """A viewset for model Item that provides `retrieve`,
+    `create`, `list`, 'update' and 'delete' actions."""
+    model = Item
+    queryset = Item.objects.all()
+    filterset_class = ItemNameFilter
+    serializer_action_classes = {
+        'list': ItemListSerializer,
+        'retrieve': ItemRetrieveSerializer,
+        'create': ItemCreateUpdateSerializer,
+        'update': ItemCreateUpdateSerializer,
+        'delete': ItemDeleteSerializer,
+        'partial_update': ItemCreateUpdateSerializer
+    }
+
+    def get_serializer_class(self):
+        return self.serializer_action_classes[self.action]
+
+
+class CurrencyViewSet(mixins.ListModelMixin,
+                      mixins.CreateModelMixin,
+                      mixins.RetrieveModelMixin,
+                      mixins.UpdateModelMixin,
+                      viewsets.GenericViewSet):
+    """A viewset for model Currency that provides `retrieve`,
+    `create`, update' and `list` actions."""
+    model = Currency
+    queryset = Currency.objects.filter_not_deleted()
```

Comment:
I think it will be good idea to split it to another manager
```python
queryset = Currency.active_objects.all()
```
and also it's a good manner to use all in queryset initialization